README.md

```diff
@@ -240,6 +243,60 @@ The `mcp-grafana` binary supports various command-line flags for configuration:
 - `--server.tls-cert-file`: Path to TLS certificate file for server HTTPS
 - `--server.tls-key-file`: Path to TLS private key file for server HTTPS
 
+### Dynamic Toolset Discovery
+
+For even more efficient context window usage, you can enable **dynamic toolset discovery** mode with the `--dynamic-toolsets` flag. In this mode, tools are not loaded at startup. Instead, clients can discover available toolsets and selectively enable only the ones they need at runtime.
+
+**Benefits:**
+- Significantly reduces initial context window usage by not loading tool descriptions upfront
+- Tools are loaded on-demand only when needed
+- Preserves context space for more important data
+
+**How it works:**
+1. Start the server with `--dynamic-toolsets` flag
+2. Use `grafana_list_toolsets` to discover available toolset categories
+3. Use `grafana_enable_toolset` to load specific toolsets (e.g., "datasource", "dashboard")
+4. The client receives a `tools/list_changed` notification and refreshes its tool list
+
+**Integration with `--enabled-tools`:**
+- No flag → all toolsets are discoverable
+- `--enabled-tools=""` → no toolsets are discoverable
+- `--enabled-tools="datasource,dashboard"` → only specified toolsets are discoverable
+
+**Example configuration for Cursor/VS Code:**
+```json
+{
+  "mcpServers": {
+    "grafana": {
+      "command": "mcp-grafana",
+      "args": ["--dynamic-toolsets"],
+      "env": {
+        "GRAFANA_URL": "http://localhost:3000",
+        "GRAFANA_SERVICE_ACCOUNT_TOKEN": "<your service account token>"
+      }
+    }
+  }
+}
+```
+
+**Limitations and Client Compatibility:**
+
+Protocol Support:
+- ✅ **stdio protocol** - Fully supported
+- ❌ **SSE (Server-Sent Events)** - Not yet supported
+- ❌ **Streamable HTTP** - Not yet supported
```

Comment:
Huh, is this because notifications aren't as well supported over streamable HTTP?